planner/core/logical_plan_builder.go

```diff
@@ -3377,38 +3374,60 @@ func (b *PlanBuilder) checkOrderByPart1(ctx context.Context, sel *ast.SelectStmt
 		// Since here doesn't require FD to check, this can take place as soon as possible.
 		// Step1:
 		isNonAggregatedQuery := false
-		resolver := colResolverForOnlyFullGroupBy{
+		resolver1 := colResolverForOnlyFullGroupBy{
```

Comment:
The `resolver1` and the `resolver2` are two different resolvers. We could use a more meaningful name.